src/main/java/com/amazonaws/services/dynamodbv2/AmazonDynamoDBLockClient.java

```diff
@@ -499,11 +565,102 @@ private LockItem upsertAndMonitorExpiredLock(AcquireLockOptions options, String
             conditionalExpression = PK_EXISTS_AND_RVN_IS_THE_SAME_CONDITION;
         }
 
-        final PutItemRequest putItemRequest = new PutItemRequest().withItem(item).withTableName(tableName).withConditionExpression(conditionalExpression)
-            .withExpressionAttributeNames(expressionAttributeNames).withExpressionAttributeValues(expressionAttributeValues);
+        if (updateExistingLockRecord) {
```

Comment:
the salient part of this feature is that it does an UpdateItem, preserving other attributes while if updateExistingLockRecord is false, it blows away other attributes with a PutItem call. Allows for greater concurrency if there are multiple writers to a LockItem besides the dynamodb-lock-client

Reply:
indicated impact in PR summary